config.go

```diff
@@ -571,3 +588,24 @@ func netName(chainParams *netparams.Params) string {
 	}
 	return chainParams.Name
 }
+
+// retrieveRootPath drops all extra characters that are not part of the root path.
+// i.e. with input http://www.mydomain.com/xxxxx, http://www.mydomain.com should
+// be returned.
+func retrieveRootPath(path string) (string, error) {
+	r, err := url.Parse(path)
+	if err != nil {
+		return "", fmt.Errorf("invalid '%s' url used. error: %v", path, err)
+	}
+
+	// If the url scheme was not found, a regex expression can be used to
+	// eliminate the unwanted part.
```

Comment:
Why the regex expression evaluation makes sense here: https://github.com/golang/go/commit/ea0d2c14f8010f31dc7b892e6a546cfdb4d8ef21

Reply:
Here are a couple of tests that may or may not display unexpected behavior when pointing to a local API server.

https://play.golang.org/p/GqrdcQE85H5